File: exercism.io/csharp/perfect-numbers/PerfectNumbers.cs
using System;

public enum Classification
{
    Perfect,
    Abundant,
    Deficient
}

public static class PerfectNumbers
{
    public static Classification Classify(int number)
    {
        if(number <= 0) { throw new ArgumentOutOfRangeException(); }
        if(number == 1) { return Classification.Deficient; }
        int sum = 1;
        for(int i = 2; i <= number / 2; i++) if (number % i == 0) sum += i;
        return sum == number ? Classification.Perfect : sum > number ? Classification.Abundant : Classification.Deficient;
    }
}

File: exercism.io/javascript/perfect-numbers/perfect-numbers.js
//
// This is only a SKELETON file for the 'Perfect Numbers' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const classify = (num) => {
  if(num <= 0) throw new Error('Classification is only possible for natural numbers.')
  if(num == 1){ return "deficient"}
  let sum = 1
  for(let i = 2; i < Math.sqrt(num); i++){
    if(num % i == 0){
      sum += i + (num / i)
    }
  }

  return sum == num ? "perfect" : sum > num ? "abundant" : "deficient"

};
